receiver/kubeletstatsreceiver/scraper.go

```diff
@@ -57,6 +57,28 @@ func newKubeletScraper(
 	metricsConfig metadata.MetricsBuilderConfig,
 	nodeName string,
 ) (scraper.Metrics, error) {
+	if EnableCPUUsageMetrics.IsEnabled() {
+		if metricsConfig.Metrics.ContainerCPUUtilization.Enabled {
```

Comment:
Based on the plan described at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885#issuecomment-2285476787 (and based on @TylerHelmuth 's [comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885#issuecomment-2284615554)) we should:

a) fail if old metrics are still enabled while the feature gate is enabled (default behaviour)
b) if feature gate is disabled (manually) then the old metrics will still be in place (by default) and the new metrics won't get added by default (unless they are enabled manually).

This will give time to users to acknowledge the change and be able to handle the migration.

Reply:
Thx for the info, adapted